packages/components/src/components/SegmentedControl/styles/segmentedControl.module.scss

```diff
@@ -0,0 +1,193 @@
+    border-radius: var(--radius);
+
+    isolation: isolate;
+
+    &:disabled {
+        background-color: var(--base-color-alt);
+    }
+}
+
+.item {
+    display: flex;
+    align-items: stretch;
+    justify-content: center;
+    user-select: none;
+
+    // Apply border radius based on the index
+    &:first-child {
+        border-top-left-radius: calc(var(--radius) - var(--line-width));
+        border-bottom-left-radius: calc(var(--radius) - var(--line-width));
+    }
+    &:nth-last-child(2) {
+        border-top-right-radius: calc(var(--radius) - var(--line-width));
+        border-bottom-right-radius: calc(var(--radius) - var(--line-width));
+    }
+
+    // Focus styles for keyboard navigation, custom rounding applied to match the active indicator
+    &:focus-visible {
+        border-radius: calc(var(--radius) - var(--line-width));
+        outline: 2px solid var(--focus-ring-color);
+        outline-offset: 2px;
+    }
+}
+
+.itemLabel {
+    display: flex;
+    flex-grow: 1;
+    align-items: center;
+    justify-content: center;
+    border-radius: inherit;
+    padding-left: sizeToken(4);
+    padding-right: sizeToken(4);
+    gap: sizeToken(2);
+    font-size: var(--body-size-medium);
+    line-height: var(--body-size-medium-line-height);
+
+    // Prevent shrinking of icons
+    & svg {
+        flex-shrink: 0;
+    }
+
+    // Hover on unselected items but not when disabled
+    .item[data-state='off']:enabled &:hover {
+        background-color: #00005506;
+    }
+}
+
+.itemLabelActive {
+    @include transition-opacity;
+
+    display: inline-flex;
+    gap: sizeToken(2);
+    align-items: center;
+    justify-content: center;
+    font-weight: 500;
+    opacity: 0;
+    transition-timing-function: ease-in;
+
+    // Color the label based on the disabled state
+    .item:disabled & {
+        color: var(--box-disabled-color-inverse);
+    }
+    .item[data-state='on'] & {
+        opacity: 1;
+        transition-timing-function: ease-out;
+    }
+}
+
+.itemLabelInactive {
+    @include transition-opacity;
+
+    position: absolute;
+    display: inline-flex;
+    gap: sizeToken(2);
+    align-items: center;
+    justify-content: center;
+    font-weight: 400;
+    opacity: 1;
+    transition-timing-function: ease-out;
+
+    // Color the label based on the disabled state
+    .item:disabled & {
+        color: var(--box-disabled-color-inverse);
+    }
+    .item[data-state='on'] & {
+        opacity: 0;
+        transition-timing-function: ease-in;
+    }
+}
+
+.activeIndicator {
+    @include transition-transform;
+
+    display: none;
+    position: absolute;
+    z-index: -1;
+    top: 0;
+    left: 0;
+    height: 100%;
+
+    pointer-events: none;
+
+    // Outline of active item
+    &::before {
+        content: '';
+        position: absolute;
+        inset: 0;
+        border-radius: calc(var(--radius) - var(--line-width));
+        background-color: #fff;
+
```

Comment:
```suggestion
```

Reply:
that's reasonable but since it's just one that is left out of the party maybe we can find a different grouping so that the space does a bit more heavy lifting?

Reply:
We can have a look at StyleLint now that we are with scss to make it automatic, wdyt?

Reply:
sounds good!